### ISSUE TITLE
Regression in 0.20.0: Nested proc using outer scope variable fails to compile

### ISSUE THREAD

**Issue by khogeland** [nim-lang/Nim#11523]
This code no longer compiles:
```nim
proc foo(): proc =
  let a = 0
  return proc(): (int, int) =
    #return (0, 0) #works
    #return (a) #works
    #result = (a, 0) #works
    return (a, 0)
discard foo()
```
0.20.0:
```bash
$ choosenim 0.20.0 && nim c -f bug.nim
      Info: Version 0.20.0 already selected
Hint: used config file '/Users/kevin/.choosenim/toolchains/nim-0.20.0/config/nim.cfg' [Conf]
Hint: system [Processing]
Hint: widestrs [Processing]
Hint: io [Processing]
Hint: bug [Processing]
/Users/travis/build/nim-lang/nightlies/nim/e7471cebae2a404f3e4239f199f5a0c422484aac/lib/system/fatal.nim(39) sysFatal
Error: unhandled exception: /Users/travis/build/nim-lang/nightlies/nim/e7471cebae2a404f3e4239f199f5a0c422484aac/compiler/lambdalifting.nim(721, 18) `n[0].kind == nkEmpty`  [AssertionError]
```
0.19.6:
```bash
$ choosenim 0.19.6 && nim c -f bug.nim
   Switched to Nim 0.19.6
Hint: used config file '/Users/kevin/.choosenim/toolchains/nim-0.19.6/config/nim.cfg' [Conf]
Hint: system [Processing]
Hint: bug [Processing]
CC: nasty_bug
CC: stdlib_system
Hint:  [Link]
Hint: operation successful (12404 lines compiled; 0.453 sec total; 16.383MiB peakmem; Debug Build) [SuccessX]
```

**Comment by narimiran** [comment]
`git bisect` shows https://github.com/nim-lang/Nim/commit/045e026d0ee4e6decc4f22f1f9cbb0bdd8bc3b45 as a commit which introduced the regression.

**Comment by petabyteboy** [comment]
Will this be backported for 0.20 or when can I expect a release with the fix?

**Comment by narimiran** [comment]
Yes, the fix for this will be part of v0.20.2, which should be out this month.
